Component dropdown mega

marcelmoreau opened the pull request:
This is a mega dropdown menu component:
- when hovering  over nav.primary > ul > li, toggle `.open` class on said list item
- `.open` class is chained to `.dropdown .mega`, which transitions a transform to make the dropdown visible
- would love any feedback/improvements to JS/SCSS,whatever

Commit:
on going mega dropdown work

Commit:
Merge branch 'gh-pages' into component-dropdown-mega

* gh-pages:
  adding background video component

# Conflicts:
#	css/styles.css
#	css/styles.css.map
#	css/styles.scss

Commit:
dropdown work

Commit:
adding dropdown component

Commit:
removing inline style

keyesatimarc commented:
Looks nice; two thoughts:

Since you're "hiding" the megadrop with a transition, display none seems like overkill and might get dinged by google. Also, it looks like its animating all properties instead of just opacity, so if you change the speed from 50ms to 5 seconds, it drops in from way up high instead of just kinda showing up where it belongs. The 50ms is so fast you can't really see it moving down and appearing, but a slower speed shows what its really up to.

marcelmoreau commented:
That initial `display: none` ensures that this does not show on small screens. I `block` it at a large breakpoint.

I believe I am only transitioning two properties: `transform` and `opacity`

mbfitz commented:
The mega dropdown doesn't work in Firefox :(   might be missing some prefixes?

marcelmoreau commented:
@mbfitz Works on my FF and also Christian's. Do you have an older version?

mbfitz commented:
It works now - not sure what was going on there. Maybe a cache issue as it worked fine after a hard refresh! Kind of agree with Christian, though. The transition is a little harsh - also - what if it dropped down form the menu and not the top of the page? Seems a little weird to come from the top and not from the parent nav. ns1.com does this, too, in pure css, if you want to take a look. Also, might be nice to add an indication that there's a dropdown menu underneath the top nav, like an arrow or chevron or something.

khamer commented:
- The transform transition is too fast
- Consider moving the transition onto `.dropdown` from `.dropdown.open` so that it'll use that transition when it closes too
- `.flex` seems like an unnecessary wrapper element to me - do we need this? can we just put display: flex on `.dropdown`? Or, if it's important to have a wrapper, let's not call it `.flex` and imply implementation like that.
- I'm not a big fan of `.item` as a class either - you're not styling to it, I wouldn't include the class in the example. Plain divs should work just fine for the example.
- Don't forget to fill out the comment in _components.scss
- _components.scss:153 - don't think you need the ampersand
- dropdown.plugin.js - this looks solid; the only thing I can pick on is you never use `$navPrimary`, so you don't need that line.

Commit:
edits to dropdown menu; elm renaming; added comments; added dropdown visual indicator

marcelmoreau commented:
- I could not (quickly) get the dropdown to slide from under the parent `header.primary` element, so it not fades in. Feel free @mbfitz or whomever to modify this
- added dropdown indicator
- renamed the `.flex` and `.item` elements
- added _components comment
- removed extraneous var in the dropdown plugin js

marcelmoreau commented:
not hovering correctly

Commit:
ensure dropdown does not appear erroneously

marcelmoreau commented:
ready to review...looking for those plus unos

mbfitz commented:
No more fade or slide-in effects? Just wanted to point out the transition isn't working - probably because it's display: none to block. If you remove the display properties and try doing visibility:hidden to visible w/ the opacity it should work. Also - you should only need the transition on one, either the .dropdown.mega or .dropdown.mega.open.

Commit:
ensuring small-screen does not show dropdown

marcelmoreau commented:
Added in an opacity effect. Actually had to use a max-width media query to get this to work correctly.

mbfitz commented:
one last thing! need to add "visibility: hidden" on .dropdown and then "visibility: visible" on .dropdown.open, otherwise you trigger the dropdown showing up while hovering below the menu.

Commit:
visibility fix for dropdown

marcelmoreau commented:
fixed

mbfitz commented:
+1!

marcelmoreau commented:
anyone else?

khamer commented:
+1

TommyChanthaboune commented on js/site.js:
```diff
@@ -1,5 +1,4 @@
-
-$('.tabs').tabs();
\ No newline at end of file
+$('nav.primary').dropdown();
```
Wrap this in a document ready

Commit:
wrapping script in doc ready

Commit:
Merge branch 'gh-pages' into component-dropdown-mega

* gh-pages:
  added new layout for components list
  fixing torso's container; moving tabs plugin function call into plugin file itself
  updated and finished sticky header component w/ edits for consistency
  finished sticky header plugin and updated structure
  working on stocky header plugin
  tackled some of the feedback form my PR - all that's left is making it into a plugin
  added components section to styles.scss
  finished sticky header component
  started on sticky header component

# Conflicts:
#	components/index.html
#	css/_components.scss
#	css/styles.css
#	css/styles.css.map
#	css/styles.scss
#	js/site.js
